hieroplatform/src/main/java/org/hiero/sketches/HLogLog.java

```diff
@@ -0,0 +1,111 @@
+package org.hiero.sketches;
+
+import org.hiero.table.api.IColumn;
+import org.hiero.table.api.IMembershipSet;
+import org.hiero.table.api.IRowIterator;
+import org.hiero.utils.HashUtil;
+
+/**
+ * A class that computes an approximation of the number of distinct elements in a column. Elements
+ * are identified via their hashcode. The class uses the HyperLogLog algorithm for large estimates
+ * and LinearCounting algorithm for small estimates.
+ */
+public class HLogLog {
+    private final int regNum; //number of registers
+    private final int logregNum;
+    private final byte[] registers;
+    private final int seed;
+
+    /**
+     * @param logRegNum the logarithm of the number of registers. should be in 4...16
+     *        Going beyond 16 would make the data structure big for no good reason.
+     *        Setting logRegNum in at 10-12 should give roughly 2% accuracy most of the time
+     */
+    public HLogLog(int logRegNum, int seed) {
+        if (!spaceValid(logRegNum))
```

Comment:
For static methods the convention is to use the class name: `HLogLog.spaceValid`